### PR TITLE
Added support for composer projects

### PR DESCRIPTION
**Pull request by JoshuaBehrens**

### CODE REVIEW

File: swdc [hunk 1]
```diff
@@ -66,13 +66,43 @@ function applyFixture()
...
+    if [ -f "$SHOPWARE_FOLDER/app/install.sh" ]; then
+        echo "true"
+    fi
+}
+
+function buildComposerProjectEnvFile()
+{
+  echo 'SHOPWARE_ENV="development"' > "$SHOPWARE_FOLDER/.env"
+  # TODO do better than this
+  echo 'SHOPWARE_VERSION="5.5.4"' >> "$SHOPWARE_FOLDER/.env"
```
Comment (shyim) [inline swdc]: That looks very wrong to me
They are not required anymore in the newst composer project version

File: swdc [hunk 2]
```diff
@@ -66,13 +66,43 @@ function applyFixture()
...
+function isComposerProject()
+{
+    if [ -f "$SHOPWARE_FOLDER/app/install.sh" ]; then
+        echo "true"
+    fi
+}
+
+function buildComposerProjectEnvFile()
+{
+  echo 'SHOPWARE_ENV="development"' > "$SHOPWARE_FOLDER/.env"
```
Comment (shyim) [inline swdc]: That is not needed.It will be set from .env file

File: swdc [hunk 3]
```diff
@@ -66,13 +66,43 @@ function applyFixture()
...
-    docker-compose run --rm -eANT_OPTS=-D"file.encoding=UTF-8" cli ant -Dapp.host=$SHOPWARE_PROJECT.dev.localhost -Ddb.host=mysql -Ddb.user=root -Ddb.password=root -Ddb.name=$SHOPWARE_PROJECT -f /var/www/html/$SHOPWARE_PROJECT/build/build.xml build-unit
-    fixHooks
-    fixDeveloperConfig
+    if [ `isComposerProject` ]; then
+      buildComposerProjectEnvFile
+      docker-compose run --rm cli mysql -h mysql -u root -proot -e "CREATE DATABASE IF NOT EXISTS $SHOPWARE_PROJECT"
+      docker-compose run --rm -w "/var/www/html/${SHOPWARE_PROJECT}" cli bash -c "composer install"
+      docker-compose run --rm cli bash -c "/var/www/html/${SHOPWARE_PROJECT}/app/install.sh"      
+    else
+      docker-compose run --rm -eANT_OPTS=-D"file.encoding=UTF-8" cli ant -Dapp.host=$SHOPWARE_PROJECT.dev.localhost -Ddb.host=mysql -Ddb.user=root -Ddb.password=root -Ddb.name=$SHOPWARE_PROJECT -f /var/www/html/$SHOPWARE_PROJECT/build/build.xml build-unit
```
Comment (shyim) [inline swdc]: You have removed that on another PR :)

File: swdc (hunk 1 above)
Comment (JoshuaBehrens) [reply swdc]: From where is the shopware retrieved otherwise? Do I need to care?

File: swdc [hunk 4]
```diff
@@ -66,13 +66,43 @@ function applyFixture()
   docker-compose run --rm cli mysql -h mysql -u root -proot $SHOPWARE_PROJECT < "${DIR}/fixtures/${FIXTURE_NAME}.sql"
 }
 
+function isComposerProject()
+{
+    if [ -f "$SHOPWARE_FOLDER/app/install.sh" ]; then
```
Comment (shyim) [inline swdc]: Maybe check only app folder. Cause the install file has been changed for a meanwhile

File: swdc (hunk 2 above)
Comment (JoshuaBehrens) [reply swdc]: Which env file?

File: swdc (hunk 3 above)
Comment (JoshuaBehrens) [reply swdc]: Nope. I did not removed them on `cli ant` commands as I suspected them to be important for ant

File: swdc (hunk 1 above)
Comment (shyim) [reply swdc]: Shopware parses it from installed version in composer https://github.com/shopware/composer-project/blob/master/app/AppKernel.php#L64

File: swdc (hunk 4 above)
Comment (JoshuaBehrens) [reply swdc]:
```suggestion
    if [ -d "$SHOPWARE_FOLDER/app" ]; then
```
Like this?

File: swdc (hunk 2 above)
Comment (shyim) [reply swdc]: nevermind i had set this globally to the image https://github.com/shyim/shopware-docker/blob/master/nginx/72/Dockerfile#L3 :)

File: swdc (hunk 1 above)
Comment (JoshuaBehrens) [reply swdc]: Nice. Todo solved.

File: swdc (hunk 2 above)
Comment (JoshuaBehrens) [reply swdc]: Fine for me. Thanks for the hint

File: swdc (hunk 4 above)
Comment (shyim) [reply swdc]: Yep

File: swdc [hunk 5]
```diff
@@ -66,13 +66,40 @@ function applyFixture()
...
+
 case "$1" in
   build)
     checkParameter
     clearCache
-    docker-compose run --rm -eANT_OPTS=-D"file.encoding=UTF-8" cli ant -Dapp.host=$SHOPWARE_PROJECT.dev.localhost -Ddb.host=mysql -Ddb.user=root -Ddb.password=root -Ddb.name=$SHOPWARE_PROJECT -f /var/www/html/$SHOPWARE_PROJECT/build/build.xml build-unit
-    fixHooks
-    fixDeveloperConfig
+    if [ `isComposerProject` ]; then
+      buildComposerProjectEnvFile
```
Comment (shyim) [inline swdc]: Move this line after composer install.
Cause shopware thinks its a update and throws a exception :)

Comment (JoshuaBehrens) [reply swdc]: I saw this as well but ignored it :see_no_evil: I will rearrange that

**Pull request merged**